src/Symfony/Component/DependencyInjection/Compiler/ResolveInvalidReferencesPass.php

```diff
@@ -101,6 +103,12 @@ private function processValue($value, int $rootLevel = 0, int $level = 0)
             if ($this->container->has($id = (string) $value)) {
                 return $value;
             }
+
+            // resolve decorated service behavior depending on decorator service
+            if ($this->currentDefinition->innerServiceId === $id && ContainerInterface::NULL_ON_INVALID_REFERENCE === $this->currentDefinition->onDecoratedInvalid) {
```

Comment:
Let's make a single property from these two and turn this check into, the 2nd check is not needed: let's just not populate the property when it should not be checked.
`if ($this->currentDefinition->innerNullableServiceId === $id) {`

Reply:
Hum, I forgot `innerServiceId` already exists, nevermind.

Reply:
`innerServiceId` is already used by `src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php:149`

I can replace `onDecoratedInvalid` by `innerNullableServiceId` but I'm not sure it's that useful